File: AppLovin MAX Demo App - ObjC/AppLovin MAX Demo App - ObjC/MAX/Native Ads/ALMAXAdPlacerCollectionViewController.m
//
//  ALMAXAdPlacerCollectionViewController.m
//  AppLovin MAX Demo App - ObjC
//
//  Created by Ritam Sarmah on 4/1/22.
//  Copyright © 2022 AppLovin Corporation. All rights reserved.
//

#import "ALMAXAdPlacerCollectionViewController.h"
#import "ALTextCollectionViewCell.h"
#import <AppLovinSDK/AppLovinSDK.h>

@interface ALMAXAdPlacerCollectionViewController()<MAAdPlacerDelegate>

@property (nonatomic, strong) MACollectionViewAdPlacer *adPlacer;
@property (nonatomic, strong) NSArray<NSString *> *data;

@end

@implementation ALMAXAdPlacerCollectionViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    self.data = [UIFont.familyNames sortedArrayUsingSelector: @selector(compare:)];
    
    MAAdPlacerSettings *settings = [MAAdPlacerSettings settingsWithAdUnitIdentifier: @"YOUR_AD_UNIT_ID"];
    [settings addFixedPosition: [NSIndexPath indexPathForItem: 2 inSection: 0]];
    [settings addFixedPosition: [NSIndexPath indexPathForItem: 8 inSection: 0]];
    settings.repeatingInterval = 5;
    
    // If using custom views, you must also set the `nativeAdViewNib` and `nativeAdViewBinder` properties on the ad placer
    
    self.adPlacer = [MACollectionViewAdPlacer placerWithCollectionView: self.collectionView settings: settings];
    self.adPlacer.delegate = self;
    [self.adPlacer loadAds];
}

#pragma mark - UICollectionViewDataSource

- (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section
{
    return self.data.count;
}

- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath
{
    ALTextCollectionViewCell *cell = (ALTextCollectionViewCell *)[collectionView al_dequeueReusableCellWithReuseIdentifier: @"ALMAXAdPlacerCollectionViewCell" forIndexPath: indexPath];
    cell.textLabel.text = self.data[indexPath.row];
    return cell;
}

#pragma mark - MAAdPlacerDelegate

- (void)didLoadAdAtIndexPath:(NSIndexPath *)indexPath {}
- (void)didRemoveAdsAtIndexPaths:(NSArray<NSIndexPath *> *)indexPaths {}
- (void)didClickAd:(MAAd *)ad {}
- (void)didPayRevenueForAd:(MAAd *)ad {}

@end

File: AppLovin MAX Demo App - Swift/AppLovin MAX Demo App - Swift/MAX/Native Ads/ALMAXAdPlacerCollectionViewController.swift
//
//  ALMAXAdPlacerCollectionViewController.swift
//  AppLovin MAX Demo App - Swift
//
//  Created by Ritam Sarmah on 4/1/22.
//  Copyright © 2022 AppLovin. All rights reserved.
//

import UIKit
import AppLovinSDK

class ALMAXAdPlacerCollectionViewController: UICollectionViewController
{
    private let data = UIFont.familyNames.sorted()
    
    private var adPlacer: MACollectionViewAdPlacer!
    
    override func viewDidLoad()
    {
        super.viewDidLoad()
        
        let settings = MAAdPlacerSettings(adUnitIdentifier: "YOUR_AD_UNIT_ID")
        settings.addFixedPosition(IndexPath(item: 2, section: 0))
        settings.addFixedPosition(IndexPath(item: 8, section: 0))
        settings.repeatingInterval = 5
        
        // If using custom views, you must also set the `nativeAdViewNib` and `nativeAdViewBinder` properties on the ad placer
        
        adPlacer = MACollectionViewAdPlacer(collectionView: collectionView, settings: settings)
        adPlacer.delegate = self
        adPlacer.loadAds()
    }
    
    // MARK: - UICollectionViewDataSource

    override func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int
    {
        return data.count
    }

    override func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell
    {
        let cell = collectionView.al_dequeueReusableCell(withReuseIdentifier: "ALMAXAdPlacerCollectionViewCell", for: indexPath) as! ALTextCollectionViewCell
        cell.textLabel.text = data[indexPath.row]
        return cell
    }
}

extension ALMAXAdPlacerCollectionViewController: MAAdPlacerDelegate
{
    func didLoadAd(at indexPath: IndexPath) {}
    
    func didRemoveAds(at indexPaths: [IndexPath]) {}

    func didClick(_ ad: MAAd) {}
    
    func didPayRevenue(for ad: MAAd) {}
}
